presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoConfigDeployer.java

```diff
@@ -78,25 +83,34 @@ public void deployConfigFiles(File configTar, Path configDir)
             throw new PrestoManagerException(format("Failed to delete contents of the directory: %s", configDir.toString()), e);
         }
         int untarResult = executor
-                .runCommand("tar", "-xzvf", configTar.getAbsolutePath(),
+                .runCommand("tar", "-xzvf", configTar.toAbsolutePath().toString(),
                         "-C", configDir.toString());
         if (untarResult != 0) {
             throw new PrestoManagerException("Failed to deploy config files", untarResult);
         }
     }
 
-    public void addConfigFiles(Path configDir, Path dataDir, Path pluginDir)
+    /**
+     * Add default Presto configuration in the given directories
+     * <p>
+     * If {@link #defaultConfig} exists, its contents will be used for
+     * the configuration directory. Otherwise, non-configurable default
+     * configuration files will be created.
+     */
+    public void writeConfigFiles(Path configDir, Path dataDir, Path pluginDir)
             throws PrestoManagerException
     {
         try {
             if (isDirectory(defaultConfig)) {
                 copyDir(defaultConfig, configDir);
-                updateProperty(configDir.resolve("node.properties"), "node.id", randomUUID().toString());
+                // TODO: Include this in install requests
```

Comment:
If we include node ID in install requests, we can more effectively guarantee the uniqueness of the IDs. Currently, there is a change (albeit a very low change) that two Agents will generate the same node ID.

If they are all generated by the Controller, the IDs are effectively guaranteed to be unique.